graphql/util.go

```diff
@@ -1290,3 +1290,63 @@ func getAPISubscriptionsForProject(ctx context.Context, projectId string) ([]*re
 	}
 	return res, nil
 }
+
+func GroupProjects(projects []model.ProjectRef, onlyDefaultedToRepo bool) ([]*GroupedProjects, error) {
+	groupsMap := make(map[string][]*restModel.APIProjectRef)
+
+	for _, p := range projects {
+		var groupName string
+		groupName = fmt.Sprintf("%s/%s", p.Owner, p.Repo)
+
+		if onlyDefaultedToRepo && p.RepoRefId == "" {
+			groupName = ""
+		}
+
+		apiProjectRef := restModel.APIProjectRef{}
+		if err := apiProjectRef.BuildFromService(p); err != nil {
+			return nil, errors.Wrap(err, "error building APIProjectRef from service")
+		}
+
+		if projs, ok := groupsMap[groupName]; ok {
+			groupsMap[groupName] = append(projs, &apiProjectRef)
+		} else {
+			groupsMap[groupName] = []*restModel.APIProjectRef{&apiProjectRef}
+		}
+	}
+
+	groupsArr := []*GroupedProjects{}
+
+	for groupName, groupedProjects := range groupsMap {
+		repoRef, err := model.FindOneRepoRef(*groupedProjects[0].RepoRefId)
+		if err != nil {
+			return nil, err
+		}
+
+		apiRepoRef := restModel.APIProjectRef{}
+		if repoRef != nil {
+			if err := apiRepoRef.BuildFromService(repoRef.ProjectRef); err != nil {
+				return nil, errors.Wrap(err, "error building APIProjectRef from service")
```

Comment:
should distinguish this error from above somehow

Reply:
done